containers/nova-flavors/Dockerfile.nova-flavors

```diff
@@ -0,0 +1,49 @@
+FROM ghcr.io/rackerlabs/understack/argo-python3.12.2-alpine3.19 AS builder
+
+# This section needs to be repeated in child images
+ARG APP_PATH=/app
+ARG APP_USER=appuser
+ARG APP_GROUP=appgroup
+ARG APP_USER_UID=1000
+ARG APP_GROUP_GID=1000
+
+
```

Comment:
This just feels really heavy in every container. Can we not just do:

```
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
          readOnlyRootFilesystem: true
          runAsNonRoot: true
```

for the pod?

if we really wanted to be slim we can use the distroless. https://github.com/GoogleContainerTools/distroless/blob/main/examples/python3/Dockerfile

Reply:
> This just feels really heavy in every container.

If the verbosity is the problem, we can remove some of these by hardcoding the
values in the commands. However having them as arguments is no "Shotgun
Surgery" when the UID needs to be adjusted or path changed.


> Can we not just do:
>   ``` securityContext:
>             allowPrivilegeEscalation: false
>             capabilities:
>               drop:
>               - ALL
>             readOnlyRootFilesystem: true
>             runAsNonRoot: true
>   ```
> for the pod?

Configuring `securityContext` in Kubernetes and setting user/group IDs in a Dockerfile are two complementary, but not identical things.

If we don't declare `USER` in a Dockerfile, the container will default to `uid=0 / root`. Now, for example, if you configured a Pod or container `securityContext` with `runAsNonRoot: true` and used such an image, the `kubelet` will simply refuse to start it.

In theory, you can extend your `securityContext` with `runAsUser: 1001` and the container will start, but that brings another set of problems:
- Application files inside the container are still owned by root, because you have not configured the `USER` during the build. This can be especially problematic for files that need to be writable (pids, logs, caches, etc.).
- Processes running inside the container run with the EUID of a user that does not exist in the system from their point of view.
- Files owned by the specified UID will not have a corresponding username displayed, causing confusion during debugging or file inspection.
- Commands like `whoami` and tools relying on user information might fail or return errors, leading to broken workflows or logging issues.
- Logs may show only numeric UIDs instead of meaningful usernames, complicating troubleshooting and security audits.
- If the same image is started in development on a developer's laptop, it will start as root, which could lead to a false sense of things working until they are pushed to Kubernetes.


> if we really wanted to be slim we can use the distroless. https://github.com/GoogleContainerTools/distroless/blob/main/examples/python3/Dockerfile

The distroless are generally larger than Alpine images. `python:3.12.2-alpine3.19` is `51.8MB` and `gcr.io/distroless/python3` is `52.8MB`. The Alpine one includes at least some of the debugging tools and have a shell. Having said that, we can certainly look into this further and maybe use the distroless ones for production as theoretically they would provide better security.

Reply:
So I think I'd be more okay with it if we pinned ourselves to something rather than making it variable and having to carry that around. Because then we'd do the security piece and the `runAsUser: 1001`.

Reply:
I'm happy to work on that change, but it will not be part of this PR as it requires changes to multiple Dockerfiles that are not related to this feature

Reply:
So this can now be rebased.

Reply:
Rebased and adjusted the `Dockerfile` as needed.